### PR TITLE
Publish-Subscribe macros support

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes: #51

### CODE REVIEW

**Comment by tomusdrw** [comment]
roborob @rphmeier to the rescue! :1st_place_medal:

File: macros/src/auto_args.rs [hunk 1]
```diff
@@ -76,21 +109,42 @@ macro_rules! build_rpc_trait {
...
+			)*
+
 		}
 	) => {
 		$(#[$t_attr])*
 		pub trait $name: Sized + Send + Sync + 'static {
-			/// Requests metadata
-			type Metadata: ::jsonrpc_core::Metadata;
+			metadata! (
+				$( $sub_name )*
```
Comment (rphmeier) [inline macros/src/auto_args.rs]: a comment explaining that traits with subscriptions have different metadata than others might make sense :)

Comment (tomusdrw) [inline macros/src/auto_args.rs]: Done.

**Comment by rphmeier** [comment]
Looks good to me, but I think we're starting to reach a tipping point w.r.t. how much this macro can handle. If we need to add more features to it, IMO it would be better to take a different tack.

**Comment by tomusdrw** [comment]
Agree. Submitted a separate issue: https://github.com/ethcore/jsonrpc/issues/66

**Pull request merged**